### PR TITLE
Use standard SPIRV-Tools transformation recipes

### PR DESCRIPTION
**Pull request by antiagainst**
Previously we manually manage the list of passes.
This commit converts to use the SPIRV-Tools standard
ones.

### CODE REVIEW

**Comment by antiagainst** [comment]
I haven't added the API parameters for invoking performance passes. I'm not sure what we should call it. `shaderc_optimization_level_one`, `shaderc_optimization_level_three`, `shaderc_optimization_level_performance`?

https://github.com/google/shaderc/blob/c9931580dbf84657796e83e61bb682df51fd72ee/libshaderc/include/shaderc/shaderc.h#L124-L127

**Comment by antiagainst** [comment]
Anyway, that can happen in a following up PR. This PR is basically internal clean up.

**Comment by ehsannas** [comment]
Looks like the bots ran before my fixes went in.

**Comment by antiagainst** [comment]
Okay, added `shader_optimization_level_performance` into the API and `-O` for `glslc`.

**Comment by chrisvarns** [comment]
Just tried this out, FYI `shaderc_compile_options_set_optimization_level` in `libshaderc/src/shaderc.cc` is missing the switch for OptimizationLevel::Performance atm.

**Comment by antiagainst** [comment]
@chrisvarns: Good catch! Thanks for pointing out! It turned out to be that we always run legalization passes for HLSL source code so my tests, using HLSL, does not reveal the problem previously. Changed to GLSL to be clear.

File: libshaderc/include/shaderc/shaderc.h [hunk 1]
```diff
@@ -122,7 +122,8 @@ typedef enum {
 
 // Optimization level.
 typedef enum {
-  shaderc_optimization_level_zero,  // no optimization
+  shaderc_optimization_level_zero,         // no optimization
+  shaderc_optimization_level_performance,  // optimize towards performance
```
Comment (dneto0) [inline libshaderc/include/shaderc/shaderc.h]: Don't insert a value in between two existing values.  That breaks compatibility with anybody who has code that is compiled against the old header. Their code might ask for size optimization and now they would end up getting performance.

File: libshaderc_util/include/libshaderc_util/compiler.h [hunk 2]
```diff
@@ -128,8 +128,9 @@ class Compiler {
 
   // Supported optimization levels.
   enum class OptimizationLevel {
-    Zero,  // No optimization.
-    Size,  // Optimization towards reducing code size.
+    Zero,         // No optimization.
+    Performance,  // Optimization towards better performance.
```
Comment (dneto0) [inline libshaderc_util/include/libshaderc_util/compiler.h]: Keep these in same order as the C header.    So reorder to match requested reorder there.

File: libshaderc/include/shaderc/shaderc.h (hunk 1 above)
Comment (antiagainst) [reply libshaderc/include/shaderc/shaderc.h]: Urgh. Yes. I don't know what I was doing...

File: libshaderc_util/include/libshaderc_util/compiler.h (hunk 2 above)
Comment (antiagainst) [reply libshaderc_util/include/libshaderc_util/compiler.h]: Fixed.

**Pull request merged**